js/initcontext.go

```diff
@@ -110,7 +119,47 @@ func newBoundInitContext(base *InitContext, ctxPtr *context.Context, rt *goja.Ru
 		programs:          programs,
 		compatibilityMode: base.compatibilityMode,
 		logger:            base.logger,
+
+		shares:     base.shares,
+		sharesLock: base.sharesLock,
+	}
+}
+
+// NewShare ...
+// TODO rename
+// TODO constructor?
+func (i *InitContext) NewShare(name string, call goja.Callable) goja.Value {
```

Comment:
I don't think it should be in a separate module, the same way `open` isn't. 
On the other hand this can be used inside the VU code .... it won't be performant, probably, but there really is no problem ... technically.
Also currently the error is a little bad if you try to use it as ... it is basically undefined, which is somewhat confusing, but so is `open`

Reply:
OK, it's fine as is too.